### ISSUE TITLE
Cannot specify the docker version of the vSphere node

### ISSUE THREAD

**Issue by kingsd041** [rancher/rancher#19141]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. I want to specify the docker version, so modify Global -> Settings -> engine-install-url to `http://ip/18.09.sh`.
2. Change the default `docker_version=18.09.3` to `docker_version=18.09.0` and upload it to `http://ip/18.09.sh`
3. Create a vSphere Node Template
4. Create a vSphere Cluster

**Result:**
The cluster can be launched, but the docker version of the node is `18.09.3`, not the expected `18.09.0`. Now I can't specify the docker version of the vSphere node.
Then I check the vSphere Node Template API and find `"engineInstallURL": null`, maybe related to this
```
{
"amazonec2Config": null,
"annotations": { },
"baseType": "nodeTemplate",
"cloudCredentialId": "cattle-global-data:vmware",
"created": "2019-03-25T11:18:29Z",
"createdTS": 1553512709000,
"creatorId": "user-nwmn8",
"driver": "vmwarevsphere",
"engineEnv": { },
"engineInstallURL": null,
"engineLabel": { },
"engineOpt": { },
"engineRegistryMirror": null,
"id": "user-nwmn8:nt-n6kj7",
```

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.2.0-rc13
- Installation option (single install/HA): single

**Comment by loganhz** [comment]
root cause is https://github.com/rancher/ui/blob/f777cf1589c3c9ec4d6015ada3e0c5d99a3e3c5d/lib/nodes/addon/components/driver-vmwarevsphere/component.js#L18

**Comment by loganhz** [comment]
It happens in 2.1.x as well. So it's not a regression.

**Comment by kingsd041** [comment]
Fixed in rancher/rancher: v2.3.0-alpha3 version, you can modify the `Docker Install URL` in the template to set the docker version.
